content/articles/Implementation-of-CI-and-CD-in-java-application(Linux)-UsingShell-and-Docker-Executor-on-GitLab/index.md

```diff
@@ -0,0 +1,192 @@
+- Here comes the executor: There will be a list of executors to choose from, and type shell (as GitLab Runner will run our system)
+
+9. Start the GitLab Runner using the following command after you've successfully registered.
+
+```
+sudo GitLab-runner start
+```
+
+
+To ensure that GitLab Runner has registered the repository and that the runner has begun. Go to GitLab Account → Select repository which you want to register with runner → Settings → CI/CD → Expand Runner. There will be a green color circle accessible, with Runners active for this project displaying a message.
+
+> If the circle is gray, the runner has not yet begun and must restart.
+
+#### Linux GitLab Runner Commands
+1. **Sudo GitLab-runner register** GitLab Runner must be used to register the project.
+2. **Sudo GitLab-runner register** Begin the runner.
+3. **Sudo GitLab-runner stop** The runner must be stopped.
+4. **Sudo GitLab-runner status** To find out how GitLab-runner is doing.
+5. **Sudo GitLab-runner unregister –name test-runner** Unregister a project's Runner and replace test-runner with your runner name, which you can find in the config.toml file (where your GitLab-runner is located).
+6. **Sudo GitLab-runner unregister –Url http://gitlab.example.com/ –token t0k3n** Runner can be removed using a URL and a token.
+7. **Sudo GitLab-runner unregister –all-runners** All Runners Should Be Unregistered
+8. **Sudo GitLab-runner restart** The GitLab Runner service is stopped and then restarted using this command.
+9. **Sudo GitLab-runner uninstall** This command disables the GitLab Runner as a service and uninstalls it.
+10. **sudo GitLab-runner exec** Run this command to get a list of available executors.
+11. **sudo GitLab-runner –help** By running the command, you can see a recent list of commands.
+12. **sudo GitLab-runner run –help** The name of the environment variable is visible.
+13. **sudo GitLab-runner –debug** To run a command in debug mode, follow these steps.
+14. **Sudo GitLab-runner exec shell** To see a complete list of she's options, go here.
+
+#### gitlab-ci.yml_shell Executor: 
+On shell executor mode, the contents of.gitlab-ci.yml are shown below. Change it as necessary, based on requirements.
+```
+stages:
+ - build
+ - execute
+
+build:
+ stage: build
+ script:  
+   - ant -f build.xml
+ 
+ artifacts:
+   paths:
+   - abc.jar
+
+execute:
+ stage: execute
+ script:  
+  - pwd  
+  - cd scripts
+  - chmod -R 777 *
+  - pwd  
+  - ./run.sh
+  ```
+
+### Java implementation on the Docker Executor
+There will be no need to manually install any software because everything will be pulled from the docker container. You can, however, install the essential software by entering the name in a yml file and exporting the path. Go to GitLab Runner Set-Up (above) and select docker instead of the shell to execute the GitLab runner in docker executor mode.
```

Comment:
```Suggestion
There will be no need to manually install any software because everything will be pulled from the docker container. You can, however, install the essential software by entering the name in a yml file and exporting the path. Go to GitLab Runner Set-Up (above) and select docker instead of the shell to execute GitLab runner in docker executor mode.
```